File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/exception/ValidationException.java
package com.windea.demo.cloudcollect.core.exception;

import com.windea.demo.cloudcollect.core.domain.enums.ResponseResult;
import org.springframework.validation.ObjectError;

import java.util.List;

/**
 * 参数校验的异常。
 */
public class ValidationException extends RuntimeException {
	private static final long serialVersionUID = 7783299109810934473L;

	private List<ObjectError> validationErrors;

	public ValidationException(List<ObjectError> validationErrors) {
		super(ResponseResult.VALIDATION_ERROR.toString());
		this.validationErrors = validationErrors;
	}

	public ValidationException(List<ObjectError> validationErrors, Throwable throwable) {
		super(ResponseResult.VALIDATION_ERROR.toString(), throwable);
		this.validationErrors = validationErrors;
	}


	public List<ObjectError> getValidationErrors() {
		return validationErrors;
	}

	public void setValidationErrors(List<ObjectError> validationErrors) {
		this.validationErrors = validationErrors;
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/exception/ValidationException.kt
package com.windea.demo.cloudcollect.core.exception

import com.windea.demo.cloudcollect.core.domain.enums.*
import org.springframework.validation.*

/**参数校验的异常。 */
class ValidationException : RuntimeException {
	var validationErrors: List<ObjectError>
	
	constructor(validationErrors: List<ObjectError>) : super(ResponseResult.VALIDATION_ERROR.message) {
		this.validationErrors = validationErrors
	}
	
	constructor(validationErrors: List<ObjectError>, throwable: Throwable) : super(ResponseResult.VALIDATION_ERROR.message, throwable) {
		this.validationErrors = validationErrors
	}
}
